### ISSUE TITLE
How to determine the mini-batch size?

### ISSUE THREAD

**Issue by kaiseu** [intel-analytics/BigDL#462]
-b: The mini-batch size. It is expected that the mini-batch size is a multiple of node_number * core_number. In this example, node_number is 1 and the mini-batch size is suggested to be set to core_number * 4

the instruction says "mini-batch size is a multiple of node_number * core_number", so why the node_number is 1 while mini-batch size is suggested to be set to core_number * 4? In fact, if only use core_number * 1 would failed with error:

2017-02-14 15:26:49 ERROR ThreadPool$:106 - Error: java.lang.IllegalArgumentException: requirement failed: only mini-batch supported (2D tensor), got 1D tensor instead
        at scala.Predef$.require(Predef.scala:233)
        at com.intel.analytics.bigdl.nn.BatchNormalization.checkInputDim(BatchNormalization.scala:84)
        at com.intel.analytics.bigdl.nn.BatchNormalization.updateOutput(BatchNormalization.scala:101)
        at com.intel.analytics.bigdl.nn.BatchNormalization.updateOutput(BatchNormalization.scala:30)
        at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:129)
        at com.intel.analytics.bigdl.nn.Sequential.updateOutput(Sequential.scala:33)

**Comment by jason-dai** [comment]
@kaiseu  In the example, mini-batch size is suggested to be core_number * 4 (when node_number is 1) for compute efficiency purpose; as long as the mini-batch size is a multiple of node_number * core_number, it should work. 

For your error, the input to BatchNormalization needs to be a 2D matrix (e.g., 1xD matrix when batch_size is 1), but a vector is provided. I wonder if you can share your example code; it seems you are not using the Optimizer.optimize() interface?

**Comment by kaiseu** [comment]
I just run the VGG example and got above error.

**Comment by jason-dai** [comment]
What's your command used to launch the example? And what's your Spark cluster configurations?

**Comment by kaiseu** [comment]
1+3 nodes cluster, each node with 64G memory and below CPUs:
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                32
On-line CPU(s) list:   0-31
Thread(s) per core:    2
Core(s) per socket:    8
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 45
Stepping:              7

spark configuration:
spark.yarn.executor.memoryOverhead=2048
spark.executor.memory=6g
spark.executor.cores=6
spark.driver.memory=4g


SparkSubmit --class com.intel.analytics.bigdl.models.vgg.Train dist/lib/bigdl-0.1.0-SNAPSHOT-jar-with-dependencies.jar --core 8 --node 3 --env spark -f CIFAR-10/cifar-10-batches-bin/ -b 96
